MDK Showcase App - MDK consuming SAP AI Core services (#58)
SAP-samples/cloud-mdk-samples: Showcase_Apps/AICoreApp/MDKGenAI.mdkproject/README.md

> # MDK Generative AI 

Reviewer: ```suggestion
# AI-Enabled Field Service App
```

> This project serves as a Demo App for consuming SAP AI Core services and SAP HANA Vector DB service with the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:

Reviewer: ```suggestion
This project serves as a demo app for consuming SAP AI Core and SAP HANA Vector DB services using the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:
```

> 1. Extracting meter readings from images.

Reviewer: ```suggestion
1. Extracting meter readings from images
```

> 2. Detecting anomalies using images.

Reviewer: ```suggestion
2. Detecting anomalies from images
```

> 3. Generating operations based on text-described anomalies.

Reviewer: ```suggestion
3. Generating operations based on text-described anomalies
```

> 4. Auto-filling forms using speech-to-text conversion.

Reviewer: ```suggestion
4. Auto-filling forms using speech-to-text conversion
```

## Getting Started
### Prerequisites

> 1. Create an instance of SAP AI Core and then create a Service Key.

Reviewer: ```suggestion
1. Create an instance of SAP AI Core and then create a service key.
```

> * refer to https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html and https://developers.sap.com/tutorials/ai-core-generative-ai.html for details on how to setup SAP AI Core and SAP AI Launchpad

Reviewer: ```suggestion
* Refer to the following tutorials for details on how to set up SAP AI Core and SAP AI Launchpad:
- https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html
- https://developers.sap.com/tutorials/ai-core-generative-ai.html

```

2. Create a deployment for a model that supports the Chat Completions API and accepts image input. This demo uses the GPT-4o model.

```bash
Name: AICoreAPI
Description: SAP AI Core deployed service
URL: <AI-API-OF-AI-CORE-SERVICE-KEY>/v2 # make sure to add /v2!
Type: HTTP
ProxyType: Internet
Authentication: OAuth2ClientCredentials
tokenServiceURL: <TOKEN-SERVICE-URL-OF-AI-CORE-SERVICE-KEY>/oauth/token
clientId: <YOUR-CLIENT-ID-OF-AI-CORE-SERVICE-KEY>
clientSecret: <YOUR-CLIENT-SECRET-OF-AI-CORE-SERVICE-KEY>
# Additional Properties:
URL.headers.AI-Resource-Group: default # adjust if necessary
URL.headers.Content-Type: application/json
HTML5.DynamicDestination: true
```
4. Create a similar destination in Mobile Services for the deployed service using the Service Key of SAP AI Core for Mobile.

Please refer to the tutorial below to create a database that supports vectors.

https://github.com/SAP-samples/cap-ai-vector-engine-sample

The testing OData service has been modified to support the following functions:

    storeEmbeddings(value: String, chunkSize: String): returns String
    deleteEmbeddings(): returns String
    getRagResponse(value: String): returns String

> For this demo, the embeddings are configured in PopulateEmbeddings.js

Reviewer: ```suggestion
For this demo, the embeddings are configured in PopulateEmbeddings.js.
```

# AI-Enabled Field Service App

This project serves as a demo app for consuming SAP AI Core and SAP HANA Vector DB services using the SAP Mobile Development Kit (MDK). It demonstrates the following use cases:

1. Extracting meter readings from images
2. Detecting anomalies from images
3. Generating operations based on text-described anomalies
4. Auto-filling forms using speech-to-text conversion

## Getting Started
### Prerequisites
1. Create an instance of SAP AI Core and then create a service key.

> * Refer to the following tutorials for details on how to set up SAP AI Core and SAP AI Launchpad:

Reviewer: ```suggestion
    > Refer to the following tutorials for details on how to set up [SAP AI Core](https://developers.sap.com/tutorials/ai-core-generative-ai.html) and [SAP AI Launchpad](https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html).
```

- https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html
- https://developers.sap.com/tutorials/ai-core-generative-ai.html

2. Create a deployment for a model that supports the Chat Completions API and accepts image input. This demo uses the GPT-4o model.

```bash
Name: AICoreAPI
Description: SAP AI Core deployed service
URL: <AI-API-OF-AI-CORE-SERVICE-KEY>/v2 # make sure to add /v2!
Type: HTTP
ProxyType: Internet
Authentication: OAuth2ClientCredentials
tokenServiceURL: <TOKEN-SERVICE-URL-OF-AI-CORE-SERVICE-KEY>/oauth/token
clientId: <YOUR-CLIENT-ID-OF-AI-CORE-SERVICE-KEY>
clientSecret: <YOUR-CLIENT-SECRET-OF-AI-CORE-SERVICE-KEY>
# Additional Properties:
URL.headers.AI-Resource-Group: default # adjust if necessary
URL.headers.Content-Type: application/json
HTML5.DynamicDestination: true
```

> 4. Create a similar destination in Mobile Services for the deployed service using the Service Key of SAP AI Core for Mobile.

Reviewer: ```suggestion
3. Create a similar destination in Mobile Services for the deployed service using the Service Key of SAP AI Core for Mobile.
```

### Prerequisites
1. Create an instance of SAP AI Core and then create a service key.

    > Refer to the following tutorials for details on how to set up [SAP AI Core](https://developers.sap.com/tutorials/ai-core-generative-ai.html) and [SAP AI Launchpad](https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html).

> - https://developers.sap.com/tutorials/ai-core-launchpad-provisioning.html

Reviewer: we dont need these two links now as these are already there as hyperlinked.

Author: removed in [c3acd3f](https://github.com/SAP-samples/cloud-mdk-samples/pull/58/commits/c3acd3f454af4c7dc67d18522961b286a5c3238f)
